File: post-sample-server/src/main/java/com/hunseong/postsample/entity/Address.java
package com.hunseong.postsample.entity;

import lombok.AllArgsConstructor;

import javax.persistence.Embeddable;
import javax.persistence.Embedded;

/**
 * Created by Hunseong on 2022/06/14
 */
@Embeddable
public class Address {

    private String street;
    private String city;
    private String zipcode;

    @Embedded
    private Geo geo;
}

File: post-sample/app/src/main/java/hunseong/com/post_practice/data/entity/Address.kt
package hunseong.com.post_practice.data.entity

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Address(
    val street: String,
    val suite: String,
    val city: String,
    val zipcode: String,
    val geo: Geo,
) : Parcelable {
    fun toStringFormat(): String {
        return String.format("%s, %s, %s", suite, street, city)
    }
}
